doozerlib/distgit.py

```diff
@@ -736,59 +736,55 @@ def build_container(
-                            raise KeyboardInterrupt()
-
-                    if len(profile["targets"]) > 1:
-                        # FIXME: Currently we don't really suppport building images against multiple targets,
-                        # or we would overwrite the image tag when pushing to the registry.
-                        # `targets` is defined as an array just because we want to keep consistency with RPM build.
-                        raise DoozerFatalError("Building images against multiple targets is not currently supported.")
-                    target = '' if not profile["targets"] else profile["targets"][0]
-                    exectools.retry(
-                        retries=(1 if self.runtime.local else retries), wait_f=wait,
-                        task_f=lambda: self._build_container(
-                            target_image, target, profile["signing_intent"], profile["repo_type"], profile["repo_list"], terminate_event,
-                            scratch, record, dry_run=dry_run))
+                    state.record_image_success(self.runtime.state[self.runtime.command], self.metadata)
+                return (self.metadata.distgit_key, self.build_status)  # do nothing more since it's local only
+            else:
+                def wait(n):
+                    self.logger.info("Async error in image build thread [attempt #{}]".format(n + 1))
+                    # No need to retry if the failure will just recur
+                    error = self._detect_permanent_build_failures(self.runtime.group_config.image_build_log_scanner)
+                    if error is not None:
+                        for match in re.finditer("No package (.*) available", error):
+                            self._add_missing_pkgs(match.group(1))
+                        raise exectools.RetryException(
+                            "Saw permanent error in build logs:\n{}\nWill not retry after {} failed attempt(s)"
+                            .format(error, n + 1)
+                        )
+                    # Brew does not handle an immediate retry correctly, wait
+                    # before trying another build, terminating if interrupted.
+                    if terminate_event.wait(timeout=5 * 60):
+                        raise KeyboardInterrupt()
+
+                if len(profile["targets"]) > 1:
+                    # FIXME: Currently we don't really suppport building images against multiple targets,
+                    # or we would overwrite the image tag when pushing to the registry.
+                    # `targets` is defined as an array just because we want to keep consistency with RPM build.
+                    raise DoozerFatalError("Building images against multiple targets is not currently supported.")
+                target = '' if not profile["targets"] else profile["targets"][0]
+                exectools.retry(
+                    retries=retries, wait_f=wait,
```

Comment:
i think we still want this, to the extent that anyone might use `doozer --local` - we don't want to retry local builds.